### DISCUSSION TITLE
How can I avoid moving my snippet from <head> to <body>?

### DISCUSSION THREAD

**Discussion by EdaCZ** [jiripudil/Naja#328] [Support] [answered]
Hi!

I have snippet inside `<head>` element (for example links for preloading, page metadata for social networks or page title...). 

When I run AJAX request which redraws this snippet, after receiving data back into the page, NAJA removes this snippet from `<head>` element and places it at the beginning of the `<body>` element. 

Is there any way how to make it stay on the original place in the code?

Thanks.

**Comment by jiripudil** [comment, marked as answer]
Hello, how does your template with the snippets look? Snippets need a wrapping element – by default it's a `<div>` which is not supposed to go into `<head>` and I suspect browsers deal with it by moving it to `<body>`. There's not much Naja can do about it, sorry. Since there is no block wrapper element that [is allowed inside the `<head>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/head#see_also), the only option I can think of is to mark the whole `<head>` as a snippet.
